Dockerfile

```diff
@@ -0,0 +1,28 @@
+ARG BASE_IMAGE=golang:1.22-bookworm
+
+FROM $BASE_IMAGE AS build
+WORKDIR /usr/local/src/gop
+COPY . .
+ENV GOPROOT=/usr/local/gop
+RUN set -eux; \
+	mkdir -p $GOPROOT/bin; \
+	cp LICENSE *.mod *.sum *.md $GOPROOT/; \
+	for PATTERN in "*.go" "*.md"; do \
+		find . -mindepth 2 -name "$PATTERN" -exec cp --parents {} $GOPROOT/ \;; \
+	done; \
```

Comment:
I suggest using the command `git ls-tree --full-tree --name-only -r HEAD | grep -v -E "^\."` for better compatibility and extensibility. Maybe another PR.

Reply:
I took the filtering rules from [`.goreleaser.yaml`](https://github.com/goplus/gop/blob/75e63a7e5017d21c70b32c277f7b7b30fd19a629/.goreleaser.yaml#L56-L62) to ensure the files copied are identical to those in the [GitHub Releases](https://github.com/goplus/gop/releases) artifacts. I'm open to using `git ls-tree` if exact match isn't necessary, and can update the PR accordingly.

Reply:
There's a script `.github/workflows/check_goreleaser_config.py` that checks the files in `.goreleaser.yaml` to ensure the list is in sync with the repo. Right now, it uses `git ls-files --error-unmatch` to filter files. It meant the files are git-tracked and not start with a ".". Using a whitelist might miss some files. And I'm planning to switch it to use `git ls-tree --name-only HEAD` to get the files list directly.